pkg/prometheus/promcfg.go

```diff
@@ -4026,6 +4026,56 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 		})
 	}
 
+	// OVHCloudSDConfigs
+	if len(sc.Spec.OVHCloudSDConfigs) > 0 {
+		configs := make([][]yaml.MapItem, len(sc.Spec.OVHCloudSDConfigs))
+		for i, config := range sc.Spec.OVHCloudSDConfigs {
+			configs[i] = append(configs[i], yaml.MapItem{
+				Key:   "application_key",
+				Value: config.ApplicationKey,
+			})
+
+			value, _ := store.GetSecretKey(ctx, sc.GetNamespace(), *config.ApplicationSecret)
```

Comment:
Wouldn't we want to handle an error here, in case there is one? 🤔

Reply:
My bad. Since we have already added the secret key to store, we can just make use of the `cacheOnlyStore` and get the key from the cache like we do for [BasicAuth](https://github.com/prometheus-operator/prometheus-operator/blob/main/pkg/prometheus/promcfg.go#L438-L463), OAuth2 etc. 
```suggestion
			value, err := store.GetSecretKey(*config.ApplicationSecret)
			if err != nil {
			            level.Error(cg.logger).Log("err", fmt.Sprintf("invalid applicationSecret ref: %s", err))
		                    return cfg 
                         }
```